artiq/master/scheduler.py

```diff
@@ -485,3 +485,11 @@ def check_pause(self, rid):
                     return False
                 return r.priority_key() > run.priority_key()
         raise KeyError("RID not found")
+
+    def get_wd(self, rid):
+        """Returns the ``wd`` attribute of the run with the specified RID."""
+        for pipeline in self._pipelines.values():
+            if rid in pipeline.pool.runs:
+                run = pipeline.pool.runs[rid]
+                return run.wd
+        raise KeyError("RID not found")
```

Comment:
Matter of taste: why not?
```python
try:
    return pipeline.pool.runs[rid].wd
except KeyError:
    raise KeyError("RID not found") [from None]
```

Reply:
I was going for consistency with `check_pause()`, but yeah there's no reason why not

Reply:
Actually, if you put this whole try/except inside the loop then I don't think it would work since this would raise a key error for every pipeline that doesn't contain that RID, even if there's one that does.

Reply:
Good point indeed, sorry. Missed one level of indentation.

Reply:
Probably a moot point anyway as it looks like this PR will be rejected for release-5, and unnecessary for master/future release-6